src/overlays/actors/ovl_Obj_Grass/z_obj_grass.c

```diff
@@ -28,41 +35,495 @@ const ActorInit Obj_Grass_InitVars = {
+    sp6C[4] = (temp_f24 - sp94) * temp_f24 * temp_f28 * spA0 * 0.0015f;
+    sp6C[5] = (temp_f28 - sp90) * temp_f26 * spA4 * spA0 * 0.0015f;
+    sp6C[6] = (temp_f26 - temp_f22) * temp_f22 * temp_f20 * temp_f16 * 0.0015f;
+    sp6C[7] = (spA4 - temp_f20) * sp94 * sp90 * temp_f16 * 0.0015f;
+
+    for (i = 0; i < ARRAY_COUNT(this->unk_2F88); i++) {
+        ptr = &this->unk_2F88[i].mf[0][0];
+
+        tempf1 = sp6C[(i + 0) & 7];
+        tempf2 = sp6C[(i + 1) & 7];
+        tempf3 = sp6C[(i + 2) & 7];
+        tempf4 = sp6C[(i + 3) & 7];
+        tempf5 = sp6C[(i + 4) & 7];
+
+        ptr[0] = sp6C[1] * 0.2f;
+        ptr[1] = tempf1;
+        ptr[2] = tempf2;
+        ptr[3] = 0.0f;
+
+        ptr[4] = tempf3;
+        ptr[5] = sp6C[0];
+        ptr[6] = tempf3;
+        ptr[7] = 0.0f;
+
+        ptr[8] = tempf4;
+        ptr[9] = tempf5;
+        ptr[10] = sp6C[3] * 0.2f;
+        ptr[11] = 0.0f;
+
+        ptr[12] = 0.0f;
+        ptr[13] = 0.0f;
+        ptr[14] = 0.0f;
+        ptr[15] = 0.0f;
+    }
+}
+
+void ObjGrass_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjGrass* this = THIS;
+
+    func_809A9790(this, globalCtx);
+    func_809A983C(this, globalCtx);
+    func_809A9DB8(this);
+}
+
+void func_809AA278(ObjGrass* this, GlobalContext* globalCtx) {
+    ObjGrassStruct1* ptr;
+    ObjGrassStruct1_1* ptr2;
+    s32 i;
+    s32 j;
+    f32 distSq;
+    f32 temp_f22;
+
+    for (i = 0; i < this->unk_2944; i++) {
+        ptr = &this->unk_144[i];
+
+        temp_f22 = Math3D_Vec3fDistSq(&ptr->unk_00, &GET_ACTIVE_CAM(globalCtx)->eye);
+
+        if ((temp_f22 < SQ(1280.0f)) && func_809A9110(globalCtx, &ptr->unk_00)) {
+            ptr->unk_FE |= 1;
+
+            for (j = 0; j < ptr->unk_FC; j++) {
+                ptr2 = &ptr->unk_0C[j];
+
+                if (ptr2->unk_0F & 4) {
+                    ptr2->unk_10 = 255;
+                    ptr2->unk_0F &= ~1;
+                } else {
+                    ptr2->unk_0F |= 1;
+                    if (temp_f22 < SQ(980.0f)) {
+                        ptr2->unk_10 = 255;
+                    } else {
+                        distSq = Math3D_Vec3fDistSq(&ptr2->unk_00, &GET_ACTIVE_CAM(globalCtx)->eye);
+                        if ((distSq <= SQ(1080.0f)) || ((ptr2->unk_0F & 8) && (distSq < SQ(1180.0f)))) {
+                            ptr2->unk_10 = 255;
+                        } else if (distSq >= SQ(1180.0f)) {
+                            ptr2->unk_10 = 0;
+                        } else {
+                            ptr2->unk_10 = (1180.0f - sqrtf(distSq)) * 2.55f;
+                        }
+                    }
+                }
+            }
+        } else {
+            ptr->unk_FE &= ~1;
+        }
+    }
+}
+
+void func_809AA54C(Actor* thisx, GlobalContext* globalCtx2) {
+    ObjGrass* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+    Lights* temp_s0;
+    ObjGrassStruct1* ptr;
+    s32 i;
+    s32 j;
+    Vec3s sp70 = { 0x0000, 0x0000, 0x0000 };
```

Comment:
I don't see any reason for the full hex
```suggestion
    Vec3s sp70 = { 0, 0, 0 };
```